classes/person.rb

```diff
@@ -1,28 +1,36 @@
 require_relative './nameable'
-
 class Person < Nameable
-  def initialize(age, name = 'Unknown', parent_permission: true)
+  attr_reader :id
+  attr_accessor :name, :age, :rentals
+
+  def initialize(age, name = 'unknown', parent_permission: true)
     super()
-    @id = Random.rand(1..2000)
+    @id = rand(1..1_000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
-  attr_reader :id
-  attr_accessor :name, :age
+  def can_use_service?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
 
   def correct_name
     @name
   end
 
-  def can_use_services?
-    of_age? || @parent_permission == true
+  def rent_book(book)
+    Rental.new(Date.today, book, self)
```

Comment:
As pointed out by the previous reviewer, `rent_book` should take `date` as a parameter because you will be giving users a choice to enter the date of their own choosing. Currently, you are setting the date yourself to the current day which is not feasible for the user. Kindly update your code accordingly and then submit it for a re-review.